### PR TITLE
enh: package.json installation of playwright

### LINKED ISSUE #49: playwright-firefox and friends always installed
playwright-firefox, playwright-chromium and playwright-webkit always get installed

```
$ node --version
v12.14.1
```
```
$ npm --version
6.13.4
```

```
$ npm ls playwright-firefox
my-app@1.0.0
└─┬ jest-playwright-preset@0.0.10
  └── playwright-firefox@0.11.1
```

```
$ npm ls playwright-webkit
my-app@1.0.0
└─┬ jest-playwright-preset@0.0.10
  └── playwright-webkit@0.11.1 
```

This is because they are declared as `optionalDependencies`: https://github.com/mmarkelov/jest-playwright/blob/v0.0.10/package.json#L48-L54

This generates inside package-lock.json:
```JSON
"jest-playwright-preset": {
  "version": "0.0.10",
  "resolved": "https://registry.npmjs.org/jest-playwright-preset/-/jest-playwright-preset-0.0.10.tgz",
  "integrity": "sha512-iE7hO1v4UEvEpXXxF/7xLEnifsbceqd6UmDm82dACKakx7JAD9T3JdJkBX2j3HGOBlDYMdBeK5wdhfnWFQXDJA==",
  "dev": true,
  "requires": {
    "jest-dev-server": "^4.4.0",
    "playwright-chromium": ">=0.11.0",
    "playwright-core": ">=0.11.0",
    "playwright-firefox": ">=0.11.0",
    "playwright-webkit": ">=0.11.0"
  }
}
```

`optionalDependencies` does not work like you think: https://docs.npmjs.com/files/package.json#optionaldependencies

> If a dependency can be used, but you would like npm to proceed if it cannot be found or fails to install, then you may put it in the optionalDependencies object.

`optionalDependencies` are like `dependencies`: **always downloaded**. the *if it cannot be found* does not seem to be respected. I know it sucks, this makes it pretty useless.

Problem is firefox, chromium and webkit are heavy downloads, specially for a CI (downloaded on each CI run).

Proposed solution: remove `optionalDependencies` and clearly states in the docs that the user should download either playwright (bundles the 3 browsers) or playwright-firefox or playwright-chromium or playwright-webkit (to be verified, this is my understanding). Maybe move playwright-core to `devDependencies`?

btw jest-puppeteer does not have any `optionalDependencies`: https://github.com/smooth-code/jest-puppeteer/blob/949027b1332e270fad78eda10fd1a92c56abe1b5/package.json

btw It's always nice to declare an empty `dependencies: {}` instead of omitting it: makes it clear when reading the `package.json`

### PR DESCRIPTION
**Pull request by mxschmitt**
Have to add docs in the README.

Closes #49

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,21 +37,19 @@
...
   },
   "lint-staged": {
-    "**/*.{js,md}": [
+    "**/*.{js,ts,md}": [
       "prettier --write"
     ],
-    "*.js": "eslint --fix"
+    "*.{js,ts}": "eslint --fix"
+  },
+  "dependencies": {
```
Comment (mxschmitt) [inline package.json]: probably better optional?

Comment (mmarkelov) [reply package.json]: I thought about moving all optional to dev dependencies

Comment (mxschmitt) [reply package.json]: Probably except jest-dev-server, because this won't be installed by the user in their repo from my understanding.

Comment (mxschmitt) [reply package.json]: for me jest-dev-server is more a dependency instead of an optional or devdependency

Comment (mmarkelov) [reply package.json]: I suppose **jest-dev-server** shouldn't be installed if there is no need to use it

Comment (mxschmitt) [reply package.json]: but how should the user decide if he wants jest-dev-server or not?
Should he install it manually on his repository? Seems weird for me for such a small package. I would include it into dependency section if you don't mind.

Comment (mmarkelov) [reply package.json]: maybe you are right. It's not critical to install  jest-dev-server. So it's ok for me

**Comment by mxschmitt** [comment]
For me the documentation in the README looks already pretty clear. That you have to either install `playwright` or the standalone playwright packages.

**Comment by mmarkelov** [comment]
I suppose that documentation should be clearer. We should think about it.
For now we can merge it

**Comment by mmarkelov** [comment]
I think we are ready to publish new version with this PR

**Comment by mxschmitt** [comment]
> I think we are ready to publish new version with this PR

sounds good 👍

**Pull request merged**